### DISCUSSION TITLE
Title: Error processing evidence (audio transcription using VOSK)

### DISCUSSION THREAD

**Discussion by danpos** [sepinf-inc/IPED#1930] [Q&A] [answered]
Good morning! I've started processing two pieces of evidence, a memory card and a flashdrive, both containing audio whose content is of interest to the police investigation, so I'm trying to use IPED's default transcription option, VOSK.

To do this, I downloaded the PT-BR model available on the project's website - vosk-model-pt-fb-v0.1.1-20220516_2113 .

When I started processing, IPED suddenly terminated with the following error:

> 2023-10-16 09:45:46 [ERROR] [app.processing.Main] Processing Error: 
> java.lang.Error: Invalid memory access
> 	at org.vosk.LibVosk.vosk_model_new(Native Method) ~[vosk-0.3.32.jar:?]
> 	at org.vosk.Model.<init>(Model.java:11) ~[vosk-0.3.32.jar:?]
> 	at iped.engine.task.transcript.VoskTranscriptTask.init(VoskTranscriptTask.java:71) ~[iped-engine-4.1.5.jar:?]
> 	at iped.engine.task.transcript.AudioTranscriptTask.init(AudioTranscriptTask.java:27) ~[iped-engine-4.1.5.jar:?]
> 	at iped.engine.core.Worker.initTasks(Worker.java:128) ~[iped-engine-4.1.5.jar:?]
> 	at iped.engine.core.Worker.<init>(Worker.java:110) ~[iped-engine-4.1.5.jar:?]
> 	at iped.engine.core.Manager.initWorkers(Manager.java:543) ~[iped-engine-4.1.5.jar:?]
> 	at iped.engine.core.Manager.process(Manager.java:265) ~[iped-engine-4.1.5.jar:?]
> 	at iped.app.processing.Main.startManager(Main.java:178) [iped.jar:?]
> 	at iped.app.processing.Main.execute(Main.java:240) [iped.jar:?]
> 	at iped.app.processing.Main.main(Main.java:308) [iped.jar:?]

I searched the forum and didn't see a similar report, so I'm posting and asking for help with the issue.

Thank you in advance for your attention and support.

**Comment by lfcnassif** [comment, marked as answer]
This is not new, please see this comment and the surrounding ones:
https://github.com/sepinf-inc/IPED/issues/248#issuecomment-1176860230

Probably it is an issue with VOSK or with the model itself (since the small model works), not with the IPED integration and ideally should be reported directly to VOSK project or to the model's author.

Also see this related thread:
https://github.com/falabrasil/kaldi-br/issues/12

**Reply by danpos** [reply]
Sorry, Nassif, I only searched here in the forum ("Discussions") and not in "Issues".

Ok, got it, I'll try Wav2Vec2 and if I have any issues, I'll report them here.

Thanks for your attention and support.

**Comment by lfcnassif** [comment]
Another comment, the official Vosk portuguese models accuracy is not good for real world audios. Although slower, I would recommend the usage of the Wav2Vec2 implementation and one of its models.

**Reply by danpos** [reply]
Good evening! Just returning to give feedback.

I'm working on an old case in which the audio files are in MP3 format and are the so-called "proibidões" and the goal was to transcribe the audio in order to search for certain keywords and thus be able to characterize the criminal conduct.

The model suggested - jonatasgrosman/wav2vec2-xls-r-1b-portuguese - showed better results than the other one suggested (Edresson/wav2vec2-large-xlsr-coraa-portuguese). I also tested it with another model - lgris/wav2vec2-large-xlsr-open-brazilian-portuguese - and for the audio files in the case it gave a "better" result than the others.

The point is that it's hard to find a model that can handle what's "sung" in these files, in fact it seems to be a "dialect" and not the Portuguese language.

I ran IPED on just 10 selected files (the total is 665 unique files), which took just over eighteen minutes on my laptop.

**Reply by lfcnassif** [reply]
Hi @danpos. If you are interested in transcribing very difficult audios, maybe the whisper algorithm (to be integrated soon) can give you better results. Take a look at this draft: https://github.com/sepinf-inc/IPED/issues/1335#issuecomment-1644745545. Of course, there will be a performance cost.

**Reply by danpos** [reply]
Good morning! Thanks for your attention and support, I'll definitely give the proposed solution a try.

Considering the question fully answered, I'll close the discussion.
